language_detector/main.py

```diff
@@ -1,9 +1,58 @@
+# -*- coding: utf-8 -*-
 """This is the entry point of the program."""
 
 from .languages import LANGUAGES
-
+import string
+import re
 
 def detect_language(text, languages):
-    """Returns the detected language of given text."""
-    # implement your solution here
-    pass
+    #solution 2
+    '''
+    regex = re.compile('[%s]' % re.escape(string.punctuation))
+    ls = regex.sub('',text).split()
+    word_counts = {}
+    freq = {}
+    for word in ls:
+        word_counts[word] = word_counts.get(word,0) + 1
+    for dict in languages:
+        count = 0
+        for key,value in word_counts.items():
+            if key in dict["common_words"]:
+                count += value
+                freq[dict["name"]] = count
+    return max(freq, key=freq.get)
+    '''
+    
+    #solution 1
+    #Set Dictionary and Variable
+    stats = {}
+    found = 0
+    regex = re.compile('[%s]' % re.escape(string.punctuation))
+    text = regex.sub('',text)
+
+    #For each word found in split of the text variable passed
+    for word in text.split():
+
+        #For each item in the length of languages list
+        for i in range(len(languages)):
```

Comment:
When you iterate through a list with a for-loop, it is more pythonic to iterate through the list items than the list indices.  So,

```python
for language in languages:
```

would be preferable over

```python
for i in range(len(languages)):
```

Then, within the for-block, you could refer to ``language`` everywhere that you now have ``languages[i]``.